File: uri/ad-hock/binary_function/1593.cpp
#include <iostream>
#include <string>

using namespace std;

string divideByTwo(string num, int &remainder) {
    string result = "";
    int carry = 0;
    
    for (int i = 0; i < num.length(); i++) {
        int current = carry * 10 + (num[i] - '0');
        result += (current / 2) + '0';
        carry = current % 2;
    }
    
    remainder = carry;
    
    int start = 0;
    while (start < result.length() && result[start] == '0') {
        start++;
    }
    
    if (start == result.length()) {
        return "0";
    }
    
    return result.substr(start);
}

int countOnes(string num) {
    int count = 0;
    
    while (num != "0") {
        int remainder;
        num = divideByTwo(num, remainder);
        if (remainder == 1) {
            count++;
        }
    }
    
    return count;
}

int main() {
    int t;
    cin >> t;
    
    while (t--) {
        string num;
        cin >> num;
        cout << countOnes(num) << endl;
    }
    
    return 0;
}

File: uri/ad-hock/binary_function/1593.rs
use std::io;

fn divide_by_two(num: &str) -> (String, i32) {
    let mut result = String::new();
    let mut carry = 0;
    
    for ch in num.chars() {
        let digit = ch.to_digit(10).unwrap() as i32;
        let current = carry * 10 + digit;
        result.push(char::from_digit((current / 2) as u32, 10).unwrap());
        carry = current % 2;
    }
    
    let remainder = carry;
    
    let trimmed = result.trim_start_matches('0');
    let final_result = if trimmed.is_empty() {
        "0".to_string()
    } else {
        trimmed.to_string()
    };
    
    (final_result, remainder)
}

fn count_ones(num: &str) -> i32 {
    let mut count = 0;
    let mut current = num.to_string();
    
    while current != "0" {
        let (new_num, remainder) = divide_by_two(&current);
        if remainder == 1 {
            count += 1;
        }
        current = new_num;
    }
    
    count
}

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let t: usize = input.trim().parse().unwrap();
    
    for _ in 0..t {
        input.clear();
        io::stdin().read_line(&mut input).unwrap();
        let num = input.trim();
        println!("{}", count_ones(num));
    }
}
